app/assets/javascripts/discourse/controllers/full-page-search.js.es6

```diff
@@ -90,7 +91,9 @@ export default Ember.Controller.extend({
     this._searchOnSortChange = false;
     if (term) {
       SortOrders.forEach(order => {
-        if (term.indexOf(order.term) > -1){
+        let term_start      = term.indexOf(order.term)
+        let char_after_term = term[term_start + (order.term || '').length]
+        if (term_start > -1 && (!char_after_term || char_after_term.match(/\s/))){
```

Comment:
This is confusing to me -- why exactly does this need to change? If there's a good reason for it I can't understand why so I'd suggest a comment explaining why.